frontend/src/features/turnstile/useTurnstile.tsx

```diff
@@ -0,0 +1,160 @@
+import { useCallback, useEffect, useRef, useState } from 'react'
+import { useIntervalWhen } from 'rooks'
+
+import { featureFlags } from '~shared/constants'
+
+import { useScript } from '~hooks/useScript'
+
+import { useFeatureFlagWithDefaults } from '~features/feature-flags/queries'
+
+type TurnstileBaseConfig = {
+  sitekey?: string
+  theme?: 'light' | 'dark' | 'auto'
+  size?: 'normal' | 'compact'
+  tabindex?: number
+  language?: string
+  execution?: 'default' | 'execute'
+  appearance?: 'always' | 'execute' | 'interaction-only'
+}
+
+interface UseTurnstileProps extends TurnstileBaseConfig {
+  // id of container to load Turnstile captcha in.
+  containerID?: string
+}
+
+type TurnstileConfig = TurnstileBaseConfig & {
+  callback?: (token: string) => void
+  'expired-callback'?: () => void
+  'error-callback'?: () => void
+}
+
+type TurnstileCallbacks = {
+  ready: (callback: () => void) => void
+  render: (containerID?: string, config?: TurnstileConfig) => string
+  reset: (widgetID: string | undefined) => void
+  execute: (widgetID: string) => void
+  getResponse: (widgetID: string) => string
+}
+
+declare global {
+  interface Window {
+    turnstile?: TurnstileCallbacks
+  }
+}
+
+// todo: check usages
```

Comment:
I think we can remove this `todo` since we are using it.

Reply:
Ack